packages/web/components/cards/main-stake-card.tsx

```diff
@@ -0,0 +1,58 @@
+import { CoinPretty } from "@keplr-wallet/unit";
+import React from "react";
+import { useTranslation } from "react-multi-lang";
+
+import { Button } from "~/components/buttons";
+import { EstimatedEarningCard } from "~/components/cards/estimated-earnings-card";
+import { GenericMainCard } from "~/components/cards/generic-main-card";
+import { StakeInfoCard } from "~/components/cards/stake-info-card";
+import { StakeTab } from "~/components/control/stake-tab";
+
+export const MainStakeCard: React.FC<{
+  inputAmount?: string;
+  setInputAmount: (amount: string | undefined) => void;
```

Comment:
We should use an AmountConfig UI Config store for managing input of OSMO tokens (passing the OSMO currency to it)

This can be done via `useAmountConfig` hook.

We may also need to pass a `FakeFeeConfig` (via `useFakeFeeConfig` hook) to subtract the gas cost of staking message to prevent the transaction from failing from running out of OSMO to pay gas.

Further, the half and max functionality is included out of the box within the amount config.